watcher/watcher.go

```diff
@@ -179,9 +189,104 @@ func (wh *WatchHandler) getImagesIDsToWlidMap() map[string][]string {
 	return wh.imagesIDToWlidsMap
 }
 
+// HandleSBOMEvents handles SBOM-related events
+//
+// Handling events is defined as dispatching scan commands that match a given
+// SBOM to the main command channel
+func (wh *WatchHandler) HandleSBOMEvents(sbomEvents <-chan watch.Event, commands chan<- *apis.Command, errorCh chan<- error) {
+	defer close(commands)
+	defer close(errorCh)
+
+	for event := range sbomEvents {
+		obj, ok := event.Object.(*spdxv1beta1.SBOMSPDXv2p3)
+		if !ok {
+			errorCh <- ErrUnsupportedObject
+			continue
+		}
+
+		// We need to schedule scans only for new SBOMs, not changes to
+		// existing ones or other operations
+		if event.Type != watch.Added {
+			continue
+		}
+
+		imageID := obj.ObjectMeta.Name
+
+		wh.imageIDsMapMutex.RLock()
+		wlids, ok := wh.imagesIDToWlidsMap[imageID]
+		wh.imageIDsMapMutex.RUnlock()
+		if !ok {
+			errorCh <- ErrUnknownImageID
+			continue
+		}
+
+		wlid := wlids[0]
+
+		commands <- getCVEScanCommand(wlid, map[string]string{})
+	}
+}
+
+func (wh *WatchHandler) getSBOMWatcher() (watch.Interface, error) {
+	return wh.storageClient.SpdxV1beta1().SBOMSPDXv2p3s("").Watch(context.TODO(), v1.ListOptions{})
+}
+
 // watch for sbom changes, and trigger scans accordingly
 func (wh *WatchHandler) SBOMWatch(ctx context.Context, sessionObjChan *chan utils.SessionObj) {
-	// TODO: implement
+	inputEvents := make(chan watch.Event)
+	commands := make(chan *apis.Command)
+	errorCh := make(chan error)
+	sbomEvents := make(<-chan watch.Event)
+
+	// The watcher is considered unavailable by default
+	sbomWatcherUnavailable := make(chan struct{})
+	go func() {
+		sbomWatcherUnavailable <- struct{}{}
+	}()
+
+	go wh.HandleSBOMEvents(inputEvents, commands, errorCh)
+
+	var watcher watch.Interface
+	var err error
+	for {
+		select {
+		case sbomEvent, ok := <-sbomEvents:
+			if ok {
+				inputEvents <- sbomEvent
+			} else {
+				sbomWatcherUnavailable <- struct{}{}
+			}
```

Comment:
thought: this repeats multiple times because we both have to handle `sbomEvents` and `errorCh` closing, and also handle the watcher being unavailable

As an alternative, we might use one more channel which clearly signals that the input `sbomEvents` channel has closed and rely on the fact that we assume it cannot be close by anything other than the watcher becoming unavailable.

But then, testing `SBOMWatch` proves difficult, since it always tries to reconcile a closed Watch.